### PR TITLE
steps/execute: add test duration to `results.yaml`

### PR DESCRIPTION
**Pull request by dkarpele**
https://github.com/psss/tmt/issues/456
duration was added to `results.yaml` for execution step.
For example:
```
tmt run plans --name /plans/features/basic
...
total: 29 tests passed
```

```
$ head -n12 /var/tmp/tmt/run-020/plans/features/basic/execute/results.yaml 
/tests/discover/filtering:
    result: pass
    log:
    - data/tests/discover/filtering/output.txt
    - data/tests/discover/filtering/journal.txt
    duration: 00:00:03
/tests/discover/modified:
    result: pass
    log:
    - data/tests/discover/modified/output.txt
    - data/tests/discover/modified/journal.txt
    duration: 00:00:05
```

duration matches with beaker variable `TESTRESULT_DURATION`

```
$ grep DURATION /var/tmp/tmt/run-020/plans/features/basic/execute/data/tests/execute/duration/TestResults
TESTRESULT_DURATION=29
$ grep -A5 execute/duration: /var/tmp/tmt/run-020/plans/features/basic/execute/results.yaml 
/tests/execute/duration:
    result: pass
    log:
    - data/tests/execute/duration/output.txt
    - data/tests/execute/duration/journal.txt
    duration: 00:00:29

```

### CODE REVIEW

File: tests/execute/basic/test.sh [hunk 1]
```diff
@@ -2,6 +2,17 @@
 . /usr/share/beakerlib/beakerlib.sh || exit 1
 
 rlJournalStart
+    function check_duration() {
+        local results=$1
+        DURATION_REGEXP="^(((([0-1][0-9])|(2[0-3])):?[0-5][0-9]:?[0-5][0-9]+$))"
+        DURATION=$(grep -A5 "good:" "$results" | grep "duration:" | awk '{print $2}')
```
Comment (lukaszachy) [inline tests/execute/basic/test.sh:8]: Could you assert currect format for all test results? It should be present every time, right?

File: tmt/base.py [hunk 2]
```diff
@@ -1311,6 +1311,7 @@ class Result(object):
         result ........... test execution result
         log .............. one or more log files
         note ............. additional result details
+        duration.......... section stating how long did the test run
```
Comment (lukaszachy) [inline tmt/base.py:1314]: I'd omit `section stating` it is kind of obvious and makes the text too long

**Review by lukaszachy — CHANGES_REQUESTED** [review]
Thank you for the PR. Few comments. In addition to them, would you mind changing the commit message according to https://tmt.readthedocs.io/en/latest/contribute.html#commits.

File: tmt/base.py (hunk 2 above)
Comment (dkarpele) [reply tmt/base.py]: I changed it to `test execution time (hh:mm:ss)`

File: tests/execute/basic/test.sh (hunk 1 above)
Comment (dkarpele) [reply tests/execute/basic/test.sh]: test updated

**Review by lukaszachy — APPROVED** [review]
LGTM. 

I think this can be squashed into single commit, if you don't mind. I'll check again tomorrow.

**Comment by dkarpele** [comment]
I agree, they should be squashed.

**Comment by lukaszachy** [comment]
Thanks. Squashed and pushed to master as 795f9dd. I hope I didn't damage anything:)

**Comment by lukaszachy** [comment]
Hm, I don't like the fact it is 'closed' but not 'merged'. @psss Is there a way around it?

**Comment by psss** [comment]
I think force pushing into the original branch should fix that. At least this works before the pull request is closed. You can try :)

**Pull request closed without merging**